oioswift/proxy/controllers/obj.py

```diff
@@ -416,6 +417,17 @@ def PUT(self, req):
             data_source = ExpectedSizeReader(data_source, req.content_length)
 
         headers = self._prepare_headers(req)
+
+        # if a sharding middleware is used, use proper cache from
+        # "root container" # and retrieve versioning is set
+        container_root = req.headers.get('X-Object-Sysmeta-Oio-Bucket-Name')
+        if container_root:
+            container_root_info = self.container_info(self.account_name,
+                                                      container_root, req)
+            ver_loc = container_root_info['sysmeta'].get('versions-location')
+            ver_val = "-1" if ver_loc else "1"
+            headers[FORCEVERSIONING_HEADER] = ver_val
```

Comment:
If there is no versioning, we should leave the default value as for the the container root.